TTCN3_Project/CANFilter/com/testingtech/playits/canfilter/Car2XEntryUpdater.java

```diff
@@ -0,0 +1,145 @@
+ *  AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+ *  ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
+ *  THIS SOFTWARE.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ *  EITHER EXPRESSED OR IMPLIED, INCLUDING ANY KIND OF IMPLIED OR
+ *  EXPRESSED WARRANTY OF NON-INFRINGEMENT OR THE IMPLIED WARRANTIES
+ *  OF MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.
+ * ----------------------------------------------------------------------------
+ */
+package com.testingtech.playits.canfilter;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketTimeoutException;
+import java.util.Date;
+import java.util.Hashtable;
+import java.util.Scanner;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class Car2XEntryUpdater extends Thread {
+
+	private static final String DEFAULT_SIMULATOR_HOST = "localhost";
+	private static final int DEFAULT_SIMULATOR_PORT = 50001;
+
+	private Hashtable<String, Car2XEntry> car2xEntries;
+	private InetSocketAddress address;
+
+	/**
+	 * Connects to the openXC simulator and updates openXC respectively obd2
+	 * values within a hash table provided as a parameter.
+	 * 
+	 * @param car2xEntries
+	 */
+	public Car2XEntryUpdater(Hashtable<String, Car2XEntry> car2xEntries) {
+		this.car2xEntries = car2xEntries;
+		this.address = new InetSocketAddress(DEFAULT_SIMULATOR_HOST,
+				DEFAULT_SIMULATOR_PORT);
+	}
+
+	/**
+	 * Connects to the openXC simulator and updates values as long as there are
+	 * updated requested (at least one entry exists).
+	 */
+	public void run() {
+		Socket socket = null;
+		Scanner scanner = null;
+		try {
+			socket = new Socket();
+			System.out
+					.println("[EntryUpdater] Connecting to openXC simulator at "
+							+ address.getHostName()
+							+ " on local port "
+							+ address.getPort());
+			socket.connect(address);
+
+			scanner = new Scanner(socket.getInputStream()).useDelimiter(String
+					.valueOf((char) 0));
+
+			while (car2xEntries.size() > 0) {
+				parseString(scanner.next());
+			}
+		} catch (SocketTimeoutException e) {
+			System.out.println("[EntryUpdater] Timeout: Could not connect to "
+					+ address.getHostName() + ":" + address.getPort());
+		} catch (IOException ioe) {
+			System.err.println("[EntryUpdater] Could not connect to "
+					+ address.getHostName() + " on port " + address.getPort());
+		} finally {
+			System.out
+					.println("[EntryUpdater] No more entries to update. Shutting down.");
+			close(scanner);
+			close(socket);
+		}
+	}
+
+	private void close(Scanner scanner) {
+		if (scanner != null) {
+			scanner.close();
+		}
+	}
+
+	private void close(Socket socket) {
+		if (socket != null) {
+			try {
+				socket.close();
+			} catch (IOException e) {
+				System.out.println("[EntryUpdater] Could not close socket: "
+						+ e.getMessage());
+			}
+		}
+	}
+
+	private void parseString(String jsonString) {
```

Comment:
String parsing doesn't seem to be the task of an updater. So Please use write another class for this.